src/client/Client.cc

```diff
@@ -1971,31 +1985,34 @@ void Client::update_metadata(std::string const &k, std::string const &v)
   Mutex::Locker l(client_lock);
   assert(initialized);
 
-  if (metadata.count(k)) {
+  auto metadata_it = metadata.find(k);
+  if (metadata_it != metadata.end()) {
     ldout(cct, 1) << __func__ << " warning, overriding metadata field '" << k
-      << "' from '" << metadata[k] << "' to '" << v << "'" << dendl;
+      << "' from '" << metadata_it->second << "' to '" << v << "'" << dendl;
   }
 
-  metadata[k] = v;
+  metadata_it->second = v;
```

Comment:
This crashes ceph-fuse on startup because metadata_it is being used outside of the check for its validity.  Please do a quick check that the MDS+client start after updating this